src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -539,10 +539,14 @@ class CVulkanLogicalDevice final : public ILogicalDevice
                 logger,
                 m_physicalDevice->getLimits().spirvVersion);
         }
-        else
+        else if(cpushader->getContentType() == asset::ICPUShader::E_CONTENT_TYPE::ECT_SPIRV)
         {
             spirv = core::smart_refctd_ptr_static_cast<asset::ICPUBuffer>(source->clone(1u));
```

Comment:
why does `CVulkanShader` need to hold onto a copy of the SPIR-V >

Reply:
I think you asked @achalpandeyy a similar question a long time ago in an old PR and I can't recall the answer but I remember you were satisfied with it.
apparently in CVulkanShader:
```c
    // We want to use the SPIR-V optimizer later.
    core::smart_refctd_ptr<asset::ICPUBuffer> m_spirv;
```

Reply:
But if we use the SPIRV optimizer sooner here then there is no need I think to hold this in CVulkanShader